apps/web/src/components/dialogs/move-topics-dialog.tsx

```diff
@@ -0,0 +1,230 @@
+  const [selected, setSelected] = useState(String);
+
+  const refreshNotebooks = useStore((store) => store.refresh);
+  const getAllNotebooks = useCallback(() => {
+    refreshNotebooks();
+    return (store.get().notebooks as Notebook[]).filter((a) => {
+      return a.type !== "header" && a.id !== id;
+    });
+  }, [refreshNotebooks]);
+
+  return (
+    <Dialog
+      isOpen={true}
+      title={"Move topic(s)"}
+      description={"You can move topics between notebooks"}
+      onClose={onClose}
+      width={"30%"}
+      positiveButton={{
+        text: "Finish",
+        disabled: !selected.length,
+        onClick: async () => {
+          await db?.notebooks?.moveTopics(selected, topics);
+          refreshNotebooks();
+          onClose(true);
+        }
+      }}
+      negativeButton={{
+        text: "Cancel",
+        onClick: onClose
+      }}
+    >
+      <Flex
+        mt={1}
+        sx={{ overflowY: "hidden", flexDirection: "column" }}
+        data-test-id="notebook-list"
+      >
+        <FilteredList
+          placeholders={{
+            empty: "Add a new notebook",
+            filter: "Filter notebooks"
+          }}
+          items={getAllNotebooks}
+          filter={(notebooks, query) =>
+            db.lookup?.notebooks(notebooks, query) || []
+          }
+          onCreateNewItem={async (title) =>
+            await db.notebooks?.add({
+              title
+            })
+          }
+          renderItem={(notebook, _index, refresh) => {
+            return (
+              <Flex
+                sx={{
+                  alignItems: "center",
+                  p: "3px",
+                  cursor: "pointer",
+                  ":hover": { bg: "hover" },
+                  justifyContent: "space-between"
+                }}
+                onClick={(e) => {
+                  console.log("143 move topic dialog", notebook);
+                  setSelected(notebook.id);
+                  refresh();
+                }}
+              >
+                <Flex>
+                  {selected === notebook.id ? (
+                    <Icon.Checkmark size={18} />
+                  ) : (
+                    <Icon.ChevronRight size={18} />
+                  )}
+                  <Text
+                    variant={"body"}
+                    sx={{ fontSize: "subtitle" }}
+                    data-test-id="title"
+                  >
+                    {notebook.title}
+                  </Text>
+                </Flex>
+              </Flex>
+            );
+          }}
+        />
+      </Flex>
+    </Dialog>
+  );
+}
+
+type FilteredListProps<T extends Item> = {
+  placeholders: { filter: string; empty: string };
+  items: () => T[];
+  filter: (items: T[], query: string) => T[];
+  onCreateNewItem: (title: string) => Promise<void>;
+  renderItem: (item: T, index: number, refresh: () => void) => JSX.Element;
+};
+
+function FilteredList<T extends Item>(props: FilteredListProps<T>) {
```

Comment:
This should be moved to its own component instead of copying it all over the place.